test/support/utils.h

```diff
@@ -97,11 +97,22 @@ expect(bool expected, bool condition, const char* file, std::int32_t line, const
 
 // Do not change signature to const T&.
 // Function must be able to detect const differences between expected and actual.
-template <typename T>
+template <typename T1, typename T2>
 void
-expect_equal_val(const T& expected, const T& actual, const char* file, std::int32_t line, const char* message)
+expect_equal_val(const T1& expected, const T2& actual, const char* file, std::int32_t line, const char* message)
 {
-    if (!(expected == actual))
+    using T = std::common_type_t<T1, T2>;
+
+    bool __is_error = false;
+    if constexpr (std::is_floating_point_v<T>)
+    {
+        const auto eps = std::numeric_limits<T>::epsilon();
+        __is_error = std::fabs(T(expected) - T(actual)) >= eps;
```

Comment:
As described at https://en.cppreference.com/w/c/numeric/math/fabs, `std::fabs` defined in header `<math.h>`.
Where is this include?

Reply:
Actually, C++ version `std::fabs` is defined in `<cmath>`.   - https://en.cppreference.com/w/cpp/header/cmath
 (`math.h` - header of C- library with `fabs` function, https://en.cppreference.com/w/c/numeric/math/fabs)
 
 At the moment `<cmath>` is included via `<complex>` implicitly, https://godbolt.org/z/1cqT148YP
 But, yes, probably, we should include `<cmath>` explicitly, because I could not find any guarantee on `cppreference` about  implicit including via `<complex>`, mentioned above.